zerver/lib/avatar.py

```diff
@@ -51,7 +52,8 @@ def _get_unversioned_avatar_url(avatar_source, email=None, realm_id=None,
             # If we can, avoid doing a database query to fetch user_profile
             hash_key = user_avatar_path_from_ids(user_profile_id, realm_id)
         else:
-            user_profile = get_user_profile_by_email(email)
+            realm = Realm.objects.get(id=realm_id)
+            user_profile = get_user(email, realm)
             hash_key = user_avatar_path(user_profile)
         return upload_backend.get_avatar_url(hash_key, medium=medium)
```

Comment:
Thinking more broadly about solving this problem, would it make sense to make `realm_id` or even `realm` a mandatory argument to this code path?  I'm thinking it is possible that could be cleaner.  The main thing we need to be careful about is that to the extent possible, we don't want these functions doesn't database queries (or even really, memcached queries) in the main case where we're accessing by `user_profile_id`, since that code path gets called in loops.